packages/react-components/react-tree/src/components/FlatTree/useFlatTree.ts

```diff
@@ -1,10 +1,94 @@
 import * as React from 'react';
 import { useRootTree } from '../../hooks/useRootTree';
-import { FlatTreeProps, FlatTreeState } from './FlatTree.types';
+import { FlatTreeProps, FlatTreeSlots, FlatTreeState } from './FlatTree.types';
+import { useFlatTreeNavigation } from './useFlatTreeNavigation';
+import { HTMLElementWalker, createHTMLElementWalker } from '../../utils/createHTMLElementWalker';
+import { useFluent_unstable } from '@fluentui/react-shared-contexts';
+import { treeItemFilter } from '../../utils/treeItemFilter';
+import { ExtractSlotProps, slot, useEventCallback, useMergedRefs } from '@fluentui/react-utilities';
+import type {
+  TreeNavigationData_unstable,
+  TreeNavigationEvent_unstable,
+  TreeOpenChangeData,
+  TreeOpenChangeEvent,
+} from '../Tree/Tree.types';
+import { useTreeContext_unstable } from '../../contexts/treeContext';
+import { useSubtree } from '../../hooks/useSubtree';
+import { createNextOpenItems, useControllableOpenItems } from '../../hooks/useControllableOpenItems';
+
+export const useFlatTree_unstable: (props: FlatTreeProps, ref: React.Ref<HTMLElement>) => FlatTreeState = (
+  props,
+  ref,
+) => {
+  const level = useTreeContext_unstable(ctx => ctx.level);
+  // as level is static, this doesn't break rule of hooks
+  // and if this becomes an issue later on, this can be easily converted
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  return level > 1 ? useSubFlatTree(props, ref) : useRootFlatTree(props, ref);
+};
+
+function useRootFlatTree(props: FlatTreeProps, ref: React.Ref<HTMLElement>): FlatTreeState {
+  const [openItems, setOpenItems] = useControllableOpenItems(props);
```

Comment:
is there any point in having an uncontrolled flat tree? users will always have to control the Flat tree anyway

Reply:
Users don't have to control the whole flat tree, they can opt to control part of it and leave the rest to internals

Reply:
By using context. On the items you don't want to control you simply access context and find out it's open state, on the items you want to control you already have the value

Reply:
Ok, I removed this from this PR. To focus this PR on the navigation part! @ling1726